CRM/MembershipExtras/Service/MembershipTypeInstalmentCalculator.php

```diff
@@ -0,0 +1,248 @@
+<?php
+
+use CRM_MembershipExtras_Service_MoneyUtilities as MoneyUtilities;
+use CRM_MembershipExtras_Service_MembershipTypeTaxAmountCalculator as MembershipTypeTaxAmountCalculator;
+use CRM_MembershipExtras_Exception_InvalidMembershipTypeInstalmentCalculator as InvalidMembershipTypeInstalmentCalculator;
+use CRM_MembershipExtras_Service_MembershipTypeDatesCalculator as MembershipTypeDatesCalculator;
+use CRM_MembershipExtras_DTO_ScheduleInstalmentAmount as ScheduleInstalmentAmount;
+
+/**
+ * Class CRM_MembershipExtras_Service_MembershipTypeInstalmentCalculator
+ */
+class CRM_MembershipExtras_Service_MembershipTypeInstalmentCalculator {
+
+  const MONTHLY = 'monthly';
+  const QUARTERLY = 'quarterly';
+  const ANNUALLY = 'annually';
+
+  const MONTHLY_INSTALMENT = 12;
+  const QUARTERLY_INSTALMENT = 4;
+  const ANNUALLY_INSTALMENT = 1;
+
+  /**
+   * @var \CRM_Member_BAO_MembershipTypeDatesCalculator
```

Comment:
you are using wrong type hinting here, it should be CRM_MembershipExtras_Service_MembershipTypeDatesCalculator